kv/kvserver: TestProposalOverhead failed

[(kv/kvserver).TestProposalOverhead failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1923272&tab=buildLog) on [master@80fdcc7277927c3dbf319b6e6eafa0e90a43e3ed](https://github.com/cockroachdb/cockroach/commits/80fdcc7277927c3dbf319b6e6eafa0e90a43e3ed):

```
=== RUN   TestProposalOverhead
I200506 06:51:50.672511 25645 util/stop/stopper.go:539  quiescing
--- FAIL: TestProposalOverhead (0.01s)
    replica_test.go:6753: storagepb.RaftCommand{
            ProposerLeaseSequence:   1,
            DeprecatedProposerLease: (*roachpb.Lease)(nil),
            MaxLeaseIndex:           0x0,
            ReplicatedEvalResult:    storagepb.ReplicatedEvalResult{
                State:                (*storagepb.ReplicaState)(nil),
                Split:                (*storagepb.Split)(nil),
                Merge:                (*storagepb.Merge)(nil),
                ComputeChecksum:      (*storagepb.ComputeChecksum)(nil),
                IsLeaseRequest:       false,
                Timestamp:            hlc.Timestamp{WallTime:1588747910671904812, Logical:1},
                DeprecatedDelta:      (*enginepb.MVCCStats)(nil),
                Delta:                enginepb.MVCCStatsDelta{ContainsEstimates:0, LastUpdateNanos:1588747910671904812, IntentAge:0, GCBytesAge:0, LiveBytes:20, LiveCount:1, KeyBytes:14, KeyCount:1, ValBytes:6, ValCount:1, IntentBytes:0, IntentCount:0, SysBytes:0, SysCount:0},
                ChangeReplicas:       (*storagepb.ChangeReplicas)(nil),
                RaftLogDelta:         0,
                AddSSTable:           (*storagepb.ReplicatedEvalResult_AddSSTable)(nil),
                SuggestedCompactions: nil,
                PrevLeaseProposal:    (*hlc.Timestamp)(nil),
            },
            WriteBatch:   (*storagepb.WriteBatch)(nil),
            LogicalOpLog: (*storagepb.LogicalOpLog)(nil),
            TraceData:    {},
        }
    replica_test.go:6777: expected overhead of 42, but found 44

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestProposalOverhead PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestProposalOverhead.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/replica_test.go
+++ b/pkg/kv/kvserver/replica_test.go
@@ -6731,7 +6731,9 @@ func TestProposalOverhead(t *testing.T) {
 	key := roachpb.Key("k")
 	var overhead uint32
 
-	cfg := TestStoreConfig(nil)
+	manual := hlc.NewManualClock(123)
+	tc := testContext{manualClock: manual}
+	cfg := TestStoreConfig(hlc.NewClock(manual.UnixNano, time.Nanosecond))
 	cfg.TestingKnobs.TestingProposalFilter =
 		func(args storagebase.ProposalFilterArgs) *roachpb.Error {
 			if len(args.Req.Requests) != 1 {
@@ -6754,12 +6756,14 @@ func TestProposalOverhead(t *testing.T) {
 			t.Logf(pretty.Sprint(args.Cmd))
 			return nil
 		}
-	tc := testContext{}
 	stopper := stop.NewStopper()
 	defer stopper.Stop(context.TODO())
 	tc.StartWithStoreConfig(t, stopper, cfg)
 
 	ba := roachpb.BatchRequest{}
+	// Use a realistic timestamp with no logical portion to guarantee that the
+	// timestamp in the proposal has a deterministic size.
+	ba.Timestamp = hlc.Timestamp{WallTime: 1588747910671904812}
 	ba.Add(&roachpb.PutRequest{
 		RequestHeader: roachpb.RequestHeader{Key: key},
 		Value:         roachpb.MakeValueFromString("v"),

